sycl/include/CL/sycl/half_type.hpp

```diff
@@ -27,11 +27,121 @@
 #define __SYCL_CONSTEXPR_ON_DEVICE
 #endif
 
+#ifndef __has_builtin
+#define __has_builtin(x) 0
+#endif
+
+#if !__has_builtin(__builtin_expect)
+#define __builtin_expect(a, b) (a)
+#endif
+
+#if __cpp_lib_bit_cast || __has_builtin(__builtin_bit_cast)
+#define __SYCL_CONSTEXPR_ON_HOST constexpr
+#else
+#define __SYCL_CONSTEXPR_ON_HOST
+#endif
```

Comment:
This macro is not dependent on `__SYCL_DEVICE_ONLY__` and therefore it will affect device part as well. Either definition should be updated or the name